overlord/snapstate/refreshhints.go

```diff
@@ -149,7 +149,7 @@ func refreshHintsFromCandidates(st *state.State, updates []*snap.Info, ignoreVal
 		}
 	}
 
-	hints := []*refreshCandidate{}
+	hints := make(map[string]*refreshCandidate)
```

Comment:
should we use len(updates) here, it seems a good guess even if there's a continue below?